src/smartptrs.h

```diff
@@ -50,7 +50,7 @@ struct DeviceSpan {
   }
 
   // Copy device memory to CPU memory and return the CPU accessible memory
-  std::span<T> CopyDeviceToCpu() {
+  std::span<T> CopyDeviceToCpu() const {
```

Comment:
How is this const? p_device_memory_ is a non const std::shared_ptr and CopyDeviceToCpu is a non const method on it?